Gem/Code/Source/Components/NetworkMatchComponent.cpp

```diff
@@ -355,10 +356,21 @@ namespace MultiplayerSample
             // move to valid respawn point
             if (NetworkTeleportCompatibleComponent* teleport = playerHandle.GetEntity()->FindComponent<NetworkTeleportCompatibleComponent>())
             {
-                AZStd::pair<Multiplayer::PrefabEntityId, AZ::Transform> entityParams = 
-                    AZ::Interface<IPlayerSpawner>::Get()->GetNextPlayerSpawn();
+                AZ::Transform respawnPoint = AZ::Transform::CreateIdentity();
+                if (auto simplePlayerSpawner = AZ::Interface<Multiplayer::ISimplePlayerSpawner>::Get())
+                {
+                    respawnPoint = simplePlayerSpawner->GetNextSpawnPoint();
+
+                    // Increment the next spawn point so any new players or respawned players don't spawn in on top of us at this location.
+                    const uint32_t spawnPointCount = simplePlayerSpawner->GetSpawnPointCount();
```

Comment:
nit: seems odd that we find out the number of spawn points and then have to mod it to set valid spawn point (which is all information fetched from and resent to the simple player spawner). Is there any reason theres not just:

```
simplePlayerSpawner->AdvanceSpawnPoint()
```

which under the hood just does this advancement ie 

```
m_nextSpawnPoint = _spawnpointindex + 1 % m_numSpawnPoints
```

Reply:
Heh, you might want to weigh in over on the companion PR then: https://github.com/o3de/o3de/pull/14756 .

Reply:
@lmbr-pip yep, I removed Advance() because it was causing too much confusion and complexity, Games can do whatever they want given just indexes if they want.